### ISSUE TITLE
ssl module delivers ssl_closed message to msgq if active false is set

### ISSUE THREAD

**Issue by vans163** [erlang/otp#4771]
**Describe the bug**
If you use :ssl.connect with active false and the remote end closes the connection. the client gets a :ssl_closed message delivered to the message queue. This has been around for a while, maybe since OTP 20.

**To Reproduce**

```elixir
{:ok, socket} = :ssl.connect('google.com', 443, [{:active, false}, :binary])
:ssl.send(socket, "GET /test\r\n\r\n\r\n")
Process.sleep(5000)
:ssl.recv(socket, 0, 1)
:ssl.recv(socket, 0, 1)
{:error, :closed} = :ssl.recv(socket, 0, 1)
:ssl_closed = receive do {:ssl_closed, _} -> :ssl_closed after 0 -> nil end
```

**Expected behavior**
:ssl_closed message should not be delivered to the message queue in active false mode.

**Affected versions**
Since OTP20 (i think)

**Comment by IngelaAndin** [comment]
I do not think you tried the latest releases. Such a bug was fixed in OTP-23.1 and also in OTP 22.3.4.2  We have no plan to fix it in older releases!

**Comment by vans163** [comment]
> I do not think you tried the latest releases. Such a bug was fixed in OTP-23.1 and also in OTP 22.3.4.2 We have no plan to fix it in older releases!

Im so sorry, seems I was on `23.0-rc1` on my laptop, let me bump myself.

**Comment by vans163** [comment]
This is not present on 23.3.1/
